includes/Media/Media_Source_Taxonomy.php

```diff
@@ -82,6 +88,47 @@ public function register(): void {
 		add_filter( 'web_stories_rest_attachment_query', [ $this, 'filter_rest_generated_media_attachments' ] );
 	}
 
+	/**
+	 * Act on site initialization.
+	 *
+	 * @since 1.29.0
+	 *
+	 * @param WP_Site $site The site being initialized.
+	 */
+	public function on_site_initialization( WP_Site $site ): void {
+		parent::on_site_initialization( $site );
+
+		$this->add_missing_terms();
+	}
+
+	/**
+	 * Act on plugin activation.
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param bool $network_wide Whether the activation was done network-wide.
+	 */
+	public function on_plugin_activation( $network_wide ): void {
+		parent::on_plugin_activation( $network_wide );
+
+		$this->add_missing_terms();
+	}
+
+	/**
+	 * Adds missing terms to the taxonomy.
+	 *
+	 * @since 1.29.0
+	 */
+	private function add_missing_terms(): void {
+		foreach ( $this->media_source->get_all() as $term ) {
+			// See https://github.com/Automattic/VIP-Coding-Standards/issues/720.
+			// phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.term_exists_term_exists
+			if ( ! term_exists( $term, $this->get_taxonomy_slug() ) ) {
```

Comment:
Instead of using `term_exists` could you just get all the terms in this taxonomy and then see if it exists in an the array? This way does one query per term. Doing a `get_terms` is one query.

Reply:
Good idea!